### PR TITLE
Update development docs referencing versions of sqlite we no longer support

### PR DESCRIPTION
**Pull request by H-Shay**
Per Oliver's comment's [here](https://github.com/matrix-org/synapse/pull/15344#discussion_r1177022003)
it appears we no longer need to treat boolean columns so specially.

### CODE REVIEW

File: changelog.d/15498.doc [hunk 1]
```diff
@@ -0,0 +1 @@
+Update development docs referencing versions of sqlite we no longer support.
```
Comment (reivilibre) [inline changelog.d/15498.doc:1]:
```suggestion
Update outdated development docs that mention restrictions in versions of SQLite that we no longer support.
```

I don't want to alarm anyone by making this sound like it changes the versions of SQLite we support silently, so I've tried to bring attention to the docs just being outdated.

**Review by reivilibre — APPROVED** [review]
always good when the docs get fixed — thanks!

File: docs/development/database_schema.md [hunk 2]
```diff
@@ -155,43 +155,11 @@ def run_upgrade(
...
- * Before SQLite 3.23, `TRUE` and `FALSE` were not recognised as constants by
-   SQLite, and the `IS [NOT] TRUE`/`IS [NOT] FALSE` operators were not
-   supported. This makes it necessary to avoid using `TRUE` and `FALSE`
-   constants in SQL commands.
-
-   For example, to insert a `TRUE` value into the database, write:
-
-   ```python
-   txn.execute("INSERT INTO tbl(col) VALUES (?)", (True, ))
-   ```
```
Comment (clokep) [inline docs/development/database_schema.md:174]: Should we file an issue to track doing this for any current queries? I don't know how easy that would be to find, but I definitely know I've written a few where I purposefully pass in `True` or `False` as parameters...

Comment (clokep) [reply docs/development/database_schema.md:174]: Yes, that's what I mean.

I feel less strongly about converting `=` to `IS` than just inlining the variables: I think reducing the number of placeholders makes the queries clearer.

Comment (H-Shay) [reply docs/development/database_schema.md:174]: Right on I'll open an issue for that.

**Pull request merged**